source/client/process_impl.cc

```diff
@@ -455,6 +471,26 @@ bool ProcessImpl::runInternal(OutputCollector& collector, const std::vector<UriP
 
     Runtime::LoaderSingleton::get().initialize(*cluster_manager_);
 
+    std::list<std::unique_ptr<Envoy::Stats::Sink>> stats_sinks;
+    for (const envoy::config::metrics::v3::StatsSink& stats_sink : bootstrap.stats_sinks()) {
+      ENVOY_LOG(info, "loading stats sink configuration in Nighthawk");
+      auto& factory =
+          Envoy::Config::Utility::getAndCheckFactory<NighthawkStatsSinkFactory>(stats_sink);
+      stats_sinks.emplace_back(factory.createStatsSink(store_root_.symbolTable()));
+    }
+    for (auto& sink : stats_sinks) {
+      store_root_.addSink(*sink);
+    }
+
+    // Default flush interval is 5s.
+    std::chrono::milliseconds stats_flush_interval = std::chrono::milliseconds(
+        Envoy::DurationUtil::durationToMilliseconds(bootstrap.stats_flush_interval()));
+
+    flush_worker_ = std::make_unique<FlushWorkerImpl>(*api_, tls_, store_root_, stats_sinks,
```

Comment:
We can skip this when we know there will be nothing to flush right? We could either opt to not do it, but we could also add a `FlushWorkerImpl` which just stubs the interface to keep the flow identical (might make testing easier too).

Reply:
Thanks Otto for notice this. Added a condition to skip creating flush_worker_ when there is no sink configured.